### PR TITLE
create endpoint get Device by id

### PR DESCRIPTION
**Pull request by PedroSoaresMoj**
create endpoint get Device by id

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppselectronicmonitoringdataplatformapi/controller/DeviceControllerTest.kt
Comment (mandanakhademi) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppselectronicmonitoringdataplatformapi/controller/DeviceControllerTest.kt:1]: Line 101: getDevicesByDeviceWearerId should be changed to getDevicesByDeviceId

Comment (mandanakhademi) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppselectronicmonitoringdataplatformapi/controller/DeviceControllerTest.kt:1]: For the test in line 105 if you don't call the function in the service to make this test passed you won't need to mock the service and you can remove line 108.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppselectronicmonitoringdataplatformapi/service/DeviceServiceTest.kt
Comment (mandanakhademi) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppselectronicmonitoringdataplatformapi/service/DeviceServiceTest.kt:1]: Line 30: deviceWearerId should be replaced with deviceId and its value should be replaced with something like "test device id"

Comment (mandanakhademi) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppselectronicmonitoringdataplatformapi/service/DeviceServiceTest.kt:1]: Line 33: deviceWearerId should be replaced with deviceId

**Comment by JazJax** [comment]
I'd say as we expect to only ever get one device with a given I'd, the service should return a single device, not a list of devices.
Essentially move the .first() function into the service method

**Review by JazJax — APPROVED** [review]
lgtm!

**Pull request merged**